### ISSUE TITLE
Backport better clang support from 17 to 15 for ICG

### ISSUE THREAD

**Issue by alexlin0** [nasa/trick#376]
We have better support for the newer clang versions in 17 than 15 for ICG.  Backport these changes.

**Comment by pbielski** [comment]
I now get errors when building with clang 3.5. Is 3.8 or higher now required?

The errors look like:
```
/Users/mydir/trick/clang+llvm-3.5.0-macosx-apple-darwin/include/clang/Basic/TokenKinds.h:25:6: error: 
      ISO C++ forbids forward references to 'enum' types
enum TokenKind : unsigned short {
     ^
```

**Comment by alexlin0** [comment]
I believe this was fixed in #378
